docs/architecture/adr-006-02-client-refactor.md

```diff
@@ -0,0 +1,170 @@
+# ADR 006: ICS-02 client refactor
+
+## Changelog
+* 2022-08-01: Initial Draft
+
+## Status
+
+Accepted and applied in v6 of ibc-go
+
+## Context
+
+During the initial development of the 02-client submodule, each light client supported (06-solomachine, 07-tendermint, 09-localhost) was referenced through hardcoding. 
+To add additional light clients, code would need to be added directly to the 02-client submodule.
+Evidently, this would likely become problematic as IBC scaled to many chains using consensus mechanisms beyond the initial supported light clients. 
+Issue [#6064](https://github.com/cosmos/cosmos-sdk/issues/6064) on the SDK addressed this problem by creating a module 02-client submodule. 
+The 02-client submodule would now interact with each light client via an interface. 
+While, this change was positive in development, increasing the flexibility and adoptability of IBC, it also opened the door to new problems. 
+
+The difficulty of generalizing light clients became apparent once changes to those light clients were required. 
+Each light client represents a different consensus algorithm which may contain a host of complexity and nauances. 
+Here are some examples of issues which arose for light clients that are not applicable to all the light clients supported (06-solomachine, 07-tendermint, 09-localhost):
+
+### Tendermint non-zero height upgrades
+
+Before the launch of IBC, it was determined that the tendermint go implementation would not be capable of supporting non-zero height upgrades.
```

Comment:
```suggestion
Before the launch of IBC, it was determined that the tendermint-go implementation would not be capable of supporting non-zero height upgrades.
```

Reply:
maybe it'd be better if I just linked the repo and put "that the golang implementation of tendermint"

Reply:
updated